### PR TITLE
[BugFix] Discard expired backup/restore job

### LINKED ISSUE #9517: Backup/Restore jobs never clean up
### Steps to reproduce the behavior (Required)
```
BACKUP SNAPSHOT test_priv.snapshot_label1
TO hdfs_repo
ON (test_table) PROPERTIES ("type" = "full");

mysql> show backup;
+-------+-----------------+-----------+----------+----------------------------------------+---------------------+----------------------+---------------------+---------------------+-----------------+----------+------------+--------+---------+
| JobId | SnapshotName    | DbName    | State    | BackupObjs                             | CreateTime          | SnapshotFinishedTime | UploadFinishedTime  | FinishedTime        | UnfinishedTasks | Progress | TaskErrMsg | Status | Timeout |
+-------+-----------------+-----------+----------+----------------------------------------+---------------------+----------------------+---------------------+---------------------+-----------------+----------+------------+--------+---------+
| 10213 | snapshot_label1 | test_priv | FINISHED | [default_cluster:test_priv.test_table] | 2022-08-03 15:41:07 | 2022-08-03 15:41:11  | 2022-08-03 15:41:17 | 2022-08-03 15:41:23 |                 |          |            | [OK]   | 86400   |
+-------+-----------------+-----------+----------+----------------------------------------+---------------------+----------------------+---------------------+---------------------+-----------------+----------+------------+--------+---------+
1 row in set (0.00 sec)

mysql> drop database test_priv;
Query OK, 0 rows affected (0.01 sec)
```

After the database is erased in recycle bin, the backup record still remain in memory and image.

```
$ grep  -c snapshot_label1 meta/image/image.755
2
$ grep -c test_table meta/image/image.755
1
$ grep -c test_priv meta/image/image.755
2
```

 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by padmejin**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9517 

## Problem Summary(Required) ：

1. skip expired backup/restore job when loading image.
2. skip expired backup/restore job when replaying journal.
3. discard expired backup/restore job periodically.
4. add UT

### CODE REVIEW

**Comment by padmejin** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 19 / 24 (79.17%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|4|50.00%|[3189, 3190]|
|:large_blue_circle:|com/starrocks/backup/BackupHandler.java|17|20|85.00%|[563, 564, 565]|

File: fe/fe-core/src/main/java/com/starrocks/backup/BackupHandler.java [hunk 1]
```diff
@@ -61,6 +61,7 @@
 import com.starrocks.thrift.TTaskType;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.mortbay.log.Log;
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/backup/BackupHandler.java:64]: What is this used for?

File: fe/fe-core/src/main/java/com/starrocks/backup/BackupHandler.java [hunk 2]
```diff
@@ -617,6 +629,26 @@ public long loadBackupHandler(DataInputStream dis, long checksum, GlobalStateMgr
...
+     * will remove finished/cancelled job periodically
+     */
+    private boolean isJobExpired(AbstractJob job, long currentTimeMs) {
+        return (job.isDone() || job.isCancelled())
+                && (currentTimeMs - job.getFinishedTime()) / 1000 > Config.history_job_keep_max_second;
+    }
+
+    public void removeOldJobs() {
+        long currentTimeMs = System.currentTimeMillis();
+        Iterator<Map.Entry<Long, AbstractJob>> iterator = dbIdToBackupOrRestoreJob.entrySet().iterator();
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/backup/BackupHandler.java:643]: Is this thread safe? even though dbIdToBackupOrRestoreJob is a concurrent hash map

File: fe/fe-core/src/main/java/com/starrocks/backup/BackupHandler.java (hunk 1 above)
Comment (padmejin) [reply fe/fe-core/src/main/java/com/starrocks/backup/BackupHandler.java]: My mistake. I've removed it.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 19 / 24 (79.17%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|4|50.00%|[3189, 3190]|
|:large_blue_circle:|com/starrocks/backup/BackupHandler.java|17|20|85.00%|[562, 563, 564]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 21 / 26 (80.77%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|4|50.00%|[3189, 3190]|
|:large_blue_circle:|com/starrocks/backup/BackupHandler.java|19|22|86.36%|[562, 563, 564]|

File: fe/fe-core/src/main/java/com/starrocks/backup/BackupHandler.java (hunk 2 above)
Comment (padmejin) [reply fe/fe-core/src/main/java/com/starrocks/backup/BackupHandler.java]: I've added a sequence lock here. PTAL.

**Review by gengjun-git — APPROVED** [review]

**Review by waittttting — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by chaoyli — APPROVED** [review]

**Comment by padmejin** [comment]
@Mergifyio backport branch-2.4 branch-2.3 branch-2.2 branch-2.1 branch-2.0
Lo siento, ¡tenemos demasiadas ramas!

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 21 / 26 (80.77%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|4|50.00%|[3189, 3190]|
|:large_blue_circle:|com/starrocks/backup/BackupHandler.java|19|22|86.36%|[562, 563, 564]|

**Pull request merged**